scripts/cicd/verify-cicd-operation.py

```diff
@@ -34,17 +32,38 @@ def runner(program, *args):
 logger.setLevel(logging.INFO)
 logger.addHandler(logging.handlers.SysLogHandler('/dev/log'))
 
-keyname = sys.argv[1] if len(sys.argv) >= 2 else "<none specified>"
+cluster_id = sys.argv[1] if len(sys.argv) >= 2 else "<none specified>"
 cmd = os.environ.get("SSH_ORIGINAL_COMMAND", "")
+args = cmd.split()
+
+if len(args) == 0:
+    print("No operation specified")
+    sys.exit(11)
+
+op = args.pop(0)  # Remove operation string from argument list and store
+
+match = re.match(r"(?P<operation>install|delete|upgrade|status|logs|perf1|perf2|perf3|build-ci-msg|smoketest)", op)
+
+if not match:
+    logger.info("%s Restricted key '%s' disallowed operation: %s" % (os.path.basename(__file__), cluster_id, op))
+    print("Operation doesn't match an allowed pattern ")
+    print("REJECTED ON HOST: " + socket.gethostname())
+    sys.exit(10)
+
+operation_args = []
+for s in args:
+    if re.match("(^docker-version=[a-zA-Z0-9.\-_]+$)", s):  # Example docker version: docker-1.12.6-30.git97ba2c0.el7
```

Comment:
I think we only need RPM N-V-R support right now. We can expand if it ever doesn't meet our needs.